Fix get output code when caching is enabled

oulgen opened the pull request:
Summary: Improve output code retrieval mechanism so that it works in the presence of cache hits.

Test Plan: ci

Differential Revision: D58429602




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

masnesral commented on torch/_inductor/utils.py:
```diff
@@ -1148,19 +1140,18 @@ def call(self, *args, **kwargs):
+        save_output_code(code)
         return DummyModule()
 
     # If FX code caching is enabled, a hit prevents getting the code.
```
you can remove this comment

masnesral commented on test/inductor/test_torchinductor.py:
```diff
@@ -10646,21 +10646,25 @@ def fn2(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+
+            # Run fn2 first since it has more restrictive bounds -- to avoid cache hit
             code2 = run_and_get_triton_code(fn2_opt, *inps)
+            code1 = run_and_get_triton_code(fn1_opt, *inps)
```
bug here? There's no `inps` list any more

oulgen replied:
this is what happens when i have to run tests on OSS but export from fbcode :/

oulgen committed:
Fix get output code when caching is enabled (#128445)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/128445

Improve output code retrieval mechanism so that it works in the presence of cache hits.

Test Plan: ci

Reviewed By: masnesral

Differential Revision: D58429602
